infrastructure/application/cloudflare/snippets/validate_jwt.js

```diff
@@ -0,0 +1,41 @@
+export default {
+  async fetch(request) {
+    const API_URL_EXT = "https://api.editor.planx.dev";
+
+    const getJWT = (request) => {
+      const authHeader = request.headers.get("authorization");
+      const authHeaderJWT = authHeader?.match(/^Bearer (\S+)$/)?.[1];
```

Comment:
`\S` is the negation of `\s` (i.e. not whitespace)? I think we are expecting a JWT token to be passed here, i.e. three URL-safe base 64 strings separated by dots? So we will definitely capture it but it would generously match other stuff too? so I wonder if we should be more specific, e.g. something like: 

```regex
^Bearer ((?:[A-Za-z0-9-_]+\.?){3})$
```

see [here](https://stackoverflow.com/questions/61802832/regex-to-match-jwt) for inspo. might be overkill tho!

Reply:
Yep this is actually more correct - the reason I didn't put this in place are (rightly or wrongly - discuss!)
 - It's being forwarded to an endpoint which is going to check the format anyway (any token not formatted `header.payload.signature` will get rejected, we don't need to check twice)
 - I'm slightly paranoid about the snippet timeout (the longest operation is very likely our DB query there though for revoked tokens!)

Reply:
@DafyddLlyr makes sense to me! regex's can indeed get quite hungry unless 🐛